node/src/accountant/dao_utils.rs

```diff
@@ -876,73 +858,120 @@ mod tests {
+            function_to_validate_row_value,
+        );
 
         assert_eq!(result, Ok(2))
     }
 
     #[test]
-    fn multi_update_rows_changed_suspects_0_if_nothing_changed() {
-        let random_collection_of_changed_data: Vec<Result<bool, _>> = vec![];
-        let iterator = random_collection_of_changed_data.into_iter();
-
-        let result = rows_changed_for_multi_row_update_sql(Ok(iterator));
+    fn update_rows_and_return_their_count_suspects_0_if_nothing_to_change() {
+        let conn = Connection::open_in_memory().unwrap();
+        create_table_with_text_id_and_single_numeric_column(&conn, &vec![]);
+        let mut returning_update_stm = conn.prepare(UPDATE_STM_WITH_RETURNING).unwrap();
+        let function_to_validate_row_value = |row: &Row| row.get::<usize, i64>(0).map(|_num| true);
+
+        let result = update_rows_and_return_their_count(
+            &mut returning_update_stm,
+            function_to_validate_row_value,
+        );
 
         assert_eq!(result, Ok(0))
     }
 
     #[test]
-    fn multi_update_rows_changed_returns_all_errors() {
-        let random_collection_of_changed_data: Vec<Result<bool, _>> = vec![
-            Err(rusqlite::Error::QueryReturnedNoRows),
-            Err(rusqlite::Error::InvalidQuery),
-        ];
-        let iterator = random_collection_of_changed_data.into_iter();
-
-        let result = rows_changed_for_multi_row_update_sql(Ok(iterator));
+    fn update_rows_and_return_their_count_returns_all_errors() {
+        let conn = Connection::open_in_memory().unwrap();
+        create_table_with_text_id_and_single_numeric_column(&conn, &vec![("'A'", 12), ("'B'", 23)]);
+        let mut returning_update_stm = conn.prepare(UPDATE_STM_WITH_RETURNING).unwrap();
+        let function_to_validate_row_value =
+            |row: &Row| row.get::<usize, String>(0).map(|_num| true);
+
+        let result = update_rows_and_return_their_count(
+            &mut returning_update_stm,
+            function_to_validate_row_value,
+        );
 
         assert_eq!(
             result,
             Err(vec![
-                rusqlite::Error::QueryReturnedNoRows,
-                rusqlite::Error::InvalidQuery
+                InvalidColumnType(0, "num".to_string(), Integer),
+                InvalidColumnType(0, "num".to_string(), Integer)
             ])
         )
     }
 
     #[test]
-    #[should_panic(expected = "query failed on params binding: InvalidParameterName(\"blah\")")]
-    fn the_first_contact_rusqlite_error_just_panics_as_it_belongs_with_the_querys_args_binding() {
-        let _ = rows_changed_for_multi_row_update_sql(Err::<
-            IntoIter<Result<bool, rusqlite::Error>>,
-            _,
-        >(
-            rusqlite::Error::InvalidParameterName("blah".to_string()),
-        ));
+    #[should_panic(expected = "query failed on params binding: InvalidParameterCount(0, 1)")]
+    fn the_first_contact_rusqlite_error_panics_as_it_always_belongs_with_query_args_binding() {
```

Comment:
`update_rows_and_return_their_count_cannot_tolerate_parameterized_statement`
...except that it probably won't be called `update_rows_and_return_their_count` anymore.